### ISSUE TITLE
Keep only relevant config metadata for each stage

### ISSUE THREAD

**Issue by moralejo** [cta-observatory/cta-lstchain#795]
Coming from #747...

Currently the whole config inside the config file is saved in the metadata of a given stage. This may contain configuration items which are not used by the stage in question, and which are different from those actually used in the earlier stages leading to the data in question. 

from @Bultako:

> We could follow two different approaches to get rid of useless config params.
> 
> use specific json files for each task/process.
> filter out the config file before saving it as metadata.
> The second option seems easier to implement now, but we have to be very careful to not filter out params that are actually involved in the process.
> 
> In this PR I took the decision to register the whole config present in memory when the script is executed, but record it only in the table is produced.

Indeed it is not trivial to be sure that some parameter is not used somewhere, so I'd rather go for task-specific json files which is also cleaner in general, as @morcuended wrote in #747

**Comment by rlopezcoto** [comment]
I agree to move to the usage of config files per tool (whenever we move all scripts to tools) as it was also suggested on [this discussion]( 
https://github.com/cta-observatory/cta-lstchain/pull/599#discussion_r589674790)

This goes against #208, but maybe the software is now at a more evolved stage to be able to single out config files. For the time being, we have to keep in mind that the full configuration written in, for example, DL1b files contain only information of the config parameters used to produce them and if we want to know what are the config parameters used for example to integrate the charge, we need to read the config of lower level files.

**Comment by morcuended** [comment]
Afaik, there are only two config files normally used:
- `onsite_camera_calibration_param.json` used in `lstchain_create_calibration_file`
- `lstchain_standard_config.json` used in R0 to DL2 steps (for both observed and simulated data processing)

After #747, are both config files logged in the corresponding metadata tables within a given data file at any level?
